File: articles/decision-quality-and-the-architecture-of-judgment/r/run_all_decision_quality_workflows.R
# run_all_decision_quality_workflows.R
# Run all R workflows for Decision Quality and the Architecture of Judgment.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "decision_quality_architecture_workflow.R",
  "process_outcome_separation.R",
  "weight_sensitivity_decision_quality.R",
  "architecture_score_profiles.R",
  "outcome_bias_report.R",
  "learning_review_tables.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R decision-quality workflows completed.")

File: articles/decision-quality-and-the-architecture-of-judgment/python/run_all_decision_quality_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Decision Quality and the Architecture of Judgment."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "process_vs_outcome_quality.py",
    "decision_quality_scorecard.py",
    "outcome_bias_diagnostics.py",
    "judgment_architecture_audit.py",
    "luck_vs_skill_simulation.py",
    "review_trigger_generator.py",
    "decision_record_exporter.py",
    "decision_quality_architecture_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python decision-quality workflows completed.")


if __name__ == "__main__":
    main()
